utils.go

```diff
@@ -163,18 +155,20 @@ func loadKey(fp string) (crypto.PrivKey, error) {
 
 // saveKey attempts to save a private key to the provided filepath
 func saveKey(priv crypto.PrivKey, fp string) (err error) {
-	f, err := os.Create(filepath.Clean(fp))
```

Comment:
The old version would overwrite an existing key (passing `O_TRUNC` to the `open` syscall). The new version will error if attempting to save over an existing key. (Behavior is checked in a unit test.)